features/landscape.feature

```diff
@@ -0,0 +1,247 @@
+        {"_schema_version": "0.1", "errors": [], "failed_services": [], "needs_reboot": false, "processed_services": ["landscape"], "result": "success", "warnings": []}
+        """
+        When I run `pro status` with sudo
+        Then stdout matches regexp:
+        """
+        landscape +yes +enabled
+        """
+        When I run `sudo pro disable landscape` with sudo
+
+        # Fail to enable with assume-yes and format json
+        When I verify that running `pro enable landscape --assume-yes --format=json -- --computer-title $behave_var{machine-name system-under-test} --account-name pro-client-qa` `with sudo` exits `1`
+        Then I will see the following on stdout:
+        """
+        {"_schema_version": "0.1", "errors": [{"additional_info": {"stderr": "Created symlink /etc/systemd/system/multi-user.target.wants/landscape-client.service \u2192 /lib/systemd/system/landscape-client.service.\nInvalid account name or registration key.", "stdout": "Please wait..."}, "message": "landscape-config command failed", "message_code": "landscape-config-failed", "service": "landscape", "type": "service"}], "failed_services": ["landscape"], "needs_reboot": false, "processed_services": [], "result": "failure", "warnings": []}
+        """
+        When I run `pro status` with sudo
+        Then stdout matches regexp:
+        """
+        landscape +yes +warning
+        """
+        Then stdout contains substring:
+        """
+        Landscape is installed and configured but not registered.
+        Run `sudo landscape-config` to register, or run `sudo pro disable landscape`
+        """
+        When I run `sudo pro disable landscape` with sudo
+
+        # cleanup
+        Then I reject all pending computers on Landscape
+        Examples: ubuntu release
+            | release | machine_type  |
+            | lunar   | lxd-container |
+
+    @series.lunar
+    @uses.config.machine_type.any
+    @uses.config.machine_type.lxd-container
+    Scenario Outline: Enable Landscape interactively
+        Given a `<release>` `<machine_type>` machine with ubuntu-advantage-tools installed
+        When I set the machine token overlay to the following yaml
+        """
+        availableResources:
+        - available: true
+          name: landscape
+        machineTokenInfo:
+          contractInfo:
+            resourceEntitlements:
+              - type: landscape
+                affordances:
+                  series:
+                  - focal
+                  - jammy
+                  - kinetic
+                  - lunar
+                  - mantic
+        """
+        When I attach `contract_token` with sudo and options `--no-auto-enable`
+
+        Then I verify that running `pro enable landscape` `as non-root` exits `1`
+        And I will see the following on stderr:
+        """
+        This command must be run as root (try using sudo).
+        """
+
+        When I run `pro enable landscape` `with sudo` and the following stdin
```

Comment:
Should we add a variation here were the user incorrectly provide some configuration options ?

Reply:
Yeah that's a good idea - I'll add it

Reply:
Added